### DISCUSSION TITLE
Routes are not sending trough JS routing API

### DISCUSSION THREAD

**Discussion by marek5816** [sulu/sulu#6890] [Q&A] [answered]
| Q | A
| --- | ---
| Sulu Version | 2.4.5
| PHP Version | 7.4.30
| DB Version | 5.5.5-10.3.35-MariaDB-1:10.3.35+maria~focal
| Browser Version | Firefox 105.0

# Actual Behavior

When you create Admin Controller, register route etc. according this tutorial https://docs.sulu.io/en/2.5/book/extend-admin.html to extend admin UI, then add remaining function `getAction, putAction, postAction, postTriggerAction, deleteAction`, the routes are generated and visible trough command `debug:router` but are not sending trough this API `http://<page domain>/admin/js/routing` (only 2 are send `GET with ID` and `GET without ID`)
This resulting into problem that you are not able to add new entities, edit them or delete, resulting into error `There is no detail route for the resourceKey "<key name>"`

# Expected Behavior

All routes should be sent trough the api url `http://<page domain>/admin/js/routing`

# Steps to Reproduce

Create files according the tutorial
- Entity
- Lists (config)
- Controller
- Routes (config)

Add to the Controller file this remaining functions `getAction, putAction, postAction, postTriggerAction, deleteAction`

Look for the routes in `debug:router` and `http://<page domain>/admin/js/routing`

**Comment by alexander-schranz** [comment, marked as answer]
@marek5816 we are only exposing the get routes matching `(.+\.)?c?get_.*`, as this are the only routes which are used by Sulu frontend which just use then REST like standard to performe this thing. You can not configure sulu to user specific routes for PUT, POST, .. currently.

If you want to expose additional route you need configure that in the `fos_js_routing` example or using `expose: true` option. See for this the fos js routing documentation: https://github.com/FriendsOfSymfony/FOSJsRoutingBundle/blob/master/Resources/doc/usage.rst

See also the config via `bin/console debug:config fos_js_routing` for current expose routes and regexes.

**Comment by marek5816** [comment]
All this informations are not listed in documentation. In tutorial is shown only `cgetAction()` and then next headline show that there are generated 5 routes, I had hard times even to find that I must add remaining 4 functions and only then are routes generated (ended asking this on slack)

The next issue I see is related with this topic. In documentation is not written anything about `expose: true`, but without this annotation the buttons will not work, because the routes will not be exposed
Even person on sulu slack that was helping me (and have some sulu icon, maybe he is from team?) written me that he have no clue what can cause this problem, so he did not know this also

My suggestion is, that it would be good to update the documentation and make this things more clear.

Also the tag can be changed from **Bug** to **DX** ?
I do not know, is good point to change title, to make it more related to this issue?

**Comment by alexander-schranz** [comment]
@marek5816 Sulu doesn't require the other 4 routes it just requires the route registered in your resource registration.

https://docs.sulu.io/en/2.5/book/extend-admin.html#configure-resources

As they normally match the regex there is nothing required here.

Keep also in mind the library is not provided by Sulu and so is not documented in the Sulu docs. All about the library can be found in its own documentation: https://github.com/FriendsOfSymfony/FOSJsRoutingBundle.

If you think something is missing in some kind of the documentation please feel free to open a pull request for the docs. But keep also in mind that we will not duplicate symfony or other libraries documentation in Sulu. So maybe adding just add a link with short sentence where you think something is missing.

**Comment by alexander-schranz** [comment]
I'm closing this in favor of a docs issue: https://github.com/sulu/sulu-docs/issues/750
